core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -497,7 +498,7 @@ class Partition(val topicPartition: TopicPartition,
 
       val leaderLog = localLogOrException
       val leaderEpochStartOffset = leaderLog.logEndOffset
-      info(s"$topicPartition starts at leader epoch ${partitionState.leaderEpoch} from " +
+      stateChangeLogger.info(s"Leader $topicPartition starts at leader epoch ${partitionState.leaderEpoch} from " +
```

Comment:
It would be useful to log isr, addingReplicas and removingReplicas too.

Reply:
I was considering adding a toString method to the `AssignmentState` class to make this easier but went without it for now